haechan45th/src/middleware/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require("jsonwebtoken");
+
+const validateToken = async (req, res, next) => {
+    try {
+        const token = req.headers.authorization;
+
+        if (!token) {
+            return res.status(403).send("A token is required for authentication");
+        }
+
+        const decoded = jwt.verify(token, config.JWT_SECRET_KEY);
+        req.user = decoded;
```

Comment:
```suggestion
        const decoded = jwt.verify(token, config.JWT_SECRET_KEY);
        
        // 3) Check if user still exists
	const user = await userService.getUserById(decoded.id)

	if (!user) {
		const error = new Error('USER_DOES_NOT_EXIST')
		error.statusCode = 404
		
		return res.status(error.statusCode).json({message: error.message})
	}
	
        req.user = user
```
서버에서 토큰을 발급해주고 해당 토큰에 대한 검증을 할 때에는 위와 같이 여전히 토큰의 user가 서버에서 유효한지 또한 검증하는 로직이 추가적으로 필요합니다.